machina/envs/env_utils.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+from collections import OrderedDict
+
+
+def flatten_to_dict(flatten_obs, dict_space, dict_keys=None):
+    if dict_keys is None:
+        dict_keys = dict_space.spaces.keys()
+    obs_dict = OrderedDict()
+    begin_index = 0
+    end_index = 0
+    for key in dict_keys:
+        origin_shape = dict_space.spaces[key].shape
+        end_index += np.prod(origin_shape)
+        dim = len(flatten_obs.shape)
+        if dim == 2:  # not rnn
+            obs_dict[key] = flatten_obs[:, begin_index:end_index].reshape(
+                flatten_obs.shape[:-1] + origin_shape)
+        elif dim == 3:  # rnn
```

Comment:
Is this if statement can support rnn with sampling for example R2D2-SAC? (timeseq, sampling, batch_size, ob_dim)
I think this statement should be written as `flatten_obs[..., begin_index:end_index]`.

Reply:
yes.
I tested rnn.
I will try it based on your idea.